schemas/2019-05-15/Microsoft.Kusto.json

```diff
@@ -179,303 +182,381 @@
+        },
+        "consumerGroup": {
+          "type": "string",
+          "description": "The iot hub consumer group."
+        },
+        "tableName": {
+          "type": "string",
+          "description": "The table where the data should be ingested. Optionally the table information can be added to each message."
+        },
+        "mappingRuleName": {
+          "type": "string",
+          "description": "The mapping rule to be used to ingest the data. Optionally the mapping information can be added to each message."
+        },
+        "dataFormat": {
+          "$ref": "#/definitions/DataFormat",
+          "description": "The data format of the message. Optionally the data format can be added to each message."
+        },
+        "eventSystemProperties": {
+          "description": "The iot hub system properties.",
+          "$ref": "#/definitions/EventSystemProperties"
+        },
+        "sharedAccessPolicyName": {
+          "type": "string",
+          "description": "The name of the shared access policy."
+        }
+      }
+    },
+    "EventGridDataConnection": {
+      "description": "Information about the event grid data connection",
+      "required": [
+        "kind",
+        "properties"
+      ],
+      "type": "object",
+      "properties": {
+        "kind": {
+          "description": "Kind of the endpoint for the data connection",
+          "enum": [
+            "EventGrid"
+          ],
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/EventGridDataConnectionProperties",
+          "description": "Event grid data connection properties"
+        }
+      }
+    },
+    "EventGridDataConnectionProperties": {
+      "description": "Class representing the Kusto event grid connection properties.",
+      "required": [
+        "storageAccountResourceId",
+        "eventHubResourceId",
+        "consumerGroup",
+        "tableName",
+        "dataFormat"
+      ],
+      "type": "object",
+      "properties": {
+        "storageAccountResourceId": {
+          "type": "string",
+          "description": "The resource ID of the storage account where the data resides."
+        },
+        "eventHubResourceId": {
+          "type": "string",
+          "description": "The resource ID where the event grid is configured to send events."
+        },
+        "consumerGroup": {
+          "type": "string",
+          "description": "The event hub consumer group."
+        },
+        "tableName": {
+          "type": "string",
+          "description": "The table where the data should be ingested. Optionally the table information can be added to each message."
+        },
+        "mappingRuleName": {
+          "type": "string",
+          "description": "The mapping rule to be used to ingest the data. Optionally the mapping information can be added to each message."
+        },
+        "dataFormat": {
```

Comment:
You need to include "oneOf" for every single property in this file to reference https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression. This allows customers to use template expressions when working with this resource type.